src/NewsService.Mappers/ResponsesMappers/NewsResponseMapper.cs

```diff
@@ -15,33 +15,48 @@ namespace LT.DigitalOffice.NewsService.Mappers.ResponsesMappers
 {
     public class NewsResponseMapper : INewsResponseMapper
     {
-        private IRequestClient<IGetUserDataRequest> _client;
+        private IRequestClient<IGetUserDataRequest> _requestClient;
         private readonly ILogger _logger;
         public NewsResponseMapper(
             [FromServices] IRequestClient<IGetUserDataRequest> client,
             ILogger<NewsResponseMapper> logger)
         {
-            _client = client;
+            _requestClient = client;
             _logger = logger;
         }
-        public NewsResponse Map(DbNews value)
+        public NewsResponse Map(DbNews dbNews)
         {
-            if (value == null)
+            if (dbNews == null)
             {
                 throw new BadRequestException();
             }
 
-            User author = new User { Id = value.Id };
-            User sender = new User { Id = value.Id };
+            User author = new User();
+            User sender = new User();
+
+            author.Id = dbNews.AuthorId;
+            sender.Id = dbNews.SenderId;
 
             try
             {
-                var authorRequest = IGetUserDataRequest.CreateObj(value.AuthorId);
-                var authorResponse = _client.GetResponse<IOperationResult<IGetUserDataResponse>>(authorRequest).Result;
+                var authorRequest = IGetUserDataRequest.CreateObj(dbNews.AuthorId);
+                var authorResponse = _requestClient.GetResponse<IOperationResult<IGetUserDataResponse>>(authorRequest).Result;
+
+                if (!authorResponse.Message.IsSuccess)
+                {
+                    _logger.LogWarning($"Can not found author. Reason: '{string.Join(',', authorResponse.Message.Errors)}'");
+                }
+
                 author.FIO = $"{authorResponse.Message.Body.LastName} {authorResponse.Message.Body.FirstName} {authorResponse.Message.Body.MiddleName}".Trim();
 
-                var senderRequest = IGetUserDataRequest.CreateObj(value.SenderId);
-                var senderResponse = _client.GetResponse<IOperationResult<IGetUserDataResponse>>(senderRequest).Result;
+                var senderRequest = IGetUserDataRequest.CreateObj(dbNews.SenderId);
+                var senderResponse = _requestClient.GetResponse<IOperationResult<IGetUserDataResponse>>(senderRequest).Result;
+
+                if (!senderResponse.Message.IsSuccess)
+                {
+                    _logger.LogWarning($"Can not found sender. Reason: '{string.Join(',', senderResponse.Message.Errors)}'");
+                }
+
                 sender.FIO = $"{senderResponse.Message.Body.LastName} {senderResponse.Message.Body.FirstName} {senderResponse.Message.Body.MiddleName}".Trim();
             }
```

Comment:
Try catches both calls through a broker. Is that what you intended?

Reply:
yes it is

Reply:
I asked Spartak. He said that it is worth trying to get the sender, since we do not know exactly what went wrong with the author, maybe there was simply no user with such an identifier

Reply:
done